### PR TITLE
Mmyster/feature/test functional file manager

### PR DESCRIPTION
**Pull request by MMyster**
Need : 

- [x] ~~#1086~~ not needed anymore

### CODE REVIEW

File: tests/functional/include/tests/ft_utils.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#pragma once
+
+#include "rtos/ThisThread.h"
+
+#include "boost/ut.hpp"
+
+using namespace std::chrono;
+using namespace boost::ut;
+
+void busy_sleep(uint64_t milliseconds)
```
Comment (ladislas) [inline tests/functional/include/tests/ft_utils.h:14]: So in the end I think this function is not useful anymore.

File: tests/functional/tests/file_manager/suite_file_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+void initializeSD()
+{
+	constexpr auto default_sd_blockdevice_frequency = uint64_t {25'000'000};
+
+	sd_blockdevice.init();
+	sd_blockdevice.frequency(default_sd_blockdevice_frequency);
+
+	fatfs.mount(&sd_blockdevice);
+}
```
Comment (ladislas) [inline tests/functional/tests/file_manager/suite_file_manager.cpp:29]: I think this should be moved to `utils.h` with the same structure as

https://github.com/leka/LekaOS/blob/ad37a1c39559572db0bf0f2603959bb99d2a9d2f/app/os/main.cpp#L92-L110

File: tests/functional/tests/file_manager/suite_file_manager.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+suite suite_file_manager_kit = [] {
+	auto file = FileManagerKit::File {};
+
+	const std::filesystem::path tmp_dir			  = "/fs/var/tmp";
```
Comment (ladislas) [inline tests/functional/tests/file_manager/suite_file_manager.cpp:34]: problem was that you were trying to create `/fs/tmp` as seen here

[`93146f1` (#1087)](https://github.com/leka/LekaOS/pull/1087/commits/93146f1225cb09c9e315ae26a633c84267b83c68#diff-58867dbb0e58ede1dd8017cc825a323ab8fbe7e19686f1a44420be6748475082R34)

It seems like it's not possible to create a new dir at the root of the filesystem and one can only use already created to directories to create more.

good news `/fs/var/tmp` is available!

File: tests/functional/tests/file_manager/suite_file_manager.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	const std::filesystem::path test_sub_dir	  = test_dir / "sub/";
+	const std::filesystem::path test_filename	  = test_dir / "test_spike_lk_filesystem_kit";
+	const std::filesystem::path test_sub_filename = test_sub_dir / "test_spike_lk_filesystem_kit";
+	auto input_data								  = std::string {"Hello Leka"};
+
+	initializeSD();
+
+	std::filesystem::remove(test_filename);
+	std::filesystem::remove(test_sub_dir);
+	std::filesystem::remove(test_dir);
```
Comment (ladislas) [inline tests/functional/tests/file_manager/suite_file_manager.cpp:45]: order is important here.

I tried using `remove_all` but it was not working.

File: tests/functional/tests/file_manager/suite_file_manager.cpp [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		expect(open >> fatal) << "Failed to open file";
+	};
+
+	"write to new file "_test = [&] {
+		auto bytes = file.write(input_data);
+
+		expect(bytes == std::size(input_data) >> fatal) << "Failed to write file";
+	};
+
+	file.close();
```
Comment (ladislas) [inline tests/functional/tests/file_manager/suite_file_manager.cpp:77]: I would remove the directories+files again here, so you know you leave the fs in a sane state.

**Review by ladislas — CHANGES_REQUESTED** [review]
It now works! 

image

File: tests/functional/include/tests/ft_utils.h (hunk 1 above)
Comment (MMyster) [reply tests/functional/include/tests/ft_utils.h:14]: Great if it works! The function may not be useful today but we can always think about it in the future

**Review by ladislas — APPROVED** [review]
Working now, ready to merge

**Review by YannLocatelli — APPROVED** [review]

**Pull request merged**